components/feature/media/src/main/java/mozilla/components/feature/media/notification/MediaNotification.kt

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.media.notification
+
+import android.app.Notification
+import android.app.PendingIntent
+import android.content.Context
+import android.graphics.Bitmap
+import androidx.core.app.NotificationCompat
+import mozilla.components.browser.session.Session
+import mozilla.components.feature.media.R
+import mozilla.components.feature.media.state.MediaState
+
+private const val NOTIFICATION_CHANNEL_ID = "Media"
+
+/**
+ * Helper to display a notification for web content playing media.
+ */
+internal class MediaNotification(
+    private val context: Context
+) {
+    /**
+     * Creates a new [Notification] for the given [state].
+     */
+    fun create(state: MediaState): Notification {
+        MediaNotificationChannel.ensureChannelExists(context)
+
+        val intent = context.packageManager.getLaunchIntentForPackage(context.packageName)
+        val pendingIntent = PendingIntent.getActivity(context, 0, intent, 0)
+
+        return NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
+            .setSmallIcon(state.getNotificationIcon())
+            .setContentTitle(state.getNotificationTitle())
+            .setContentText(state.getNotificationDescription())
+            .setContentIntent(pendingIntent)
+            .setLargeIcon(state.getNotificationLargeIcon())
+            .setPriority(NotificationCompat.PRIORITY_LOW)
+            .build()
+    }
+}
+
+private fun MediaState.getNotificationTitle(): String {
+    return when (this) {
+        is MediaState.Playing -> session.titleOrUrl
```

Comment:
I can't merge them because then Kotlin doesn't know about the `session` property anymore (It's unable to offer access to `session` even though it exists in both - but not all `MediaState`s).

Reply:
Now with the data class there's only one `when` statement though!